resource.go

```diff
@@ -58,8 +62,8 @@ type Resource struct {
 }
 
 func New(config Config) (*Resource, error) {
-	if config.ClusterEndpointFunc == nil {
```

Comment:
What happened to the `ClusterEndpointFunc` check? Is diff weird?

Reply:
I was too fast. Brought it back.